### ISSUE TITLE
LinkerPreserve

### ISSUE THREAD

**Issue by jzeferino** [akavache/Akavache#295]
Is the `AkavacheSqliteLinkerOverride.cs` file with the following code mandatory in every Xamarin Forms Project?

```
    [Preserve]
    public static class LinkerPreserve
    {
        static LinkerPreserve()
        {
            throw new Exception(typeof(SQLitePersistentBlobCache).FullName);
        }
    }


    public class PreserveAttribute : Attribute
    {
    }
```

I ask this because the class has this comment, but all the projects that import the nuget have this file.
```
// Note: This class file is *required* for iOS to work correctly, and is 
// also a good idea for Android if you enable "Link All Assemblies".
```

**Comment by jzeferino** [comment]
@flagbug noted that you contributed to this repo. Can you help here?

**Comment by flagbug** [comment]
Dunno, just leave it in there, it won't hurt

**Comment by jzeferino** [comment]
Yes, but if they aren't need I could make a PR to don't generate that classes for android and PCL.

**Comment by escamoteur** [comment]
Actually I think it's a good idea to have it there. I'm struggling now for some hours to get my App working with full linkage on Android and it just does not work, even that I have this file in my project.

**Comment by jzeferino** [comment]
@escamoteur what config are you using un Release mode?

**Comment by escamoteur** [comment]
What do you mean with config? Btw. I'm on the RxUI Slack

**Comment by jzeferino** [comment]
i've already send an invite to slack channel. talk you there. please remove your message and i will remove mine from here. @escamoteur

**Comment by EmilAlipiev** [comment]
@escamoteur have you ever found a proper solution for full linking

**Comment by escamoteur** [comment]
I'm no longer using akavache in my projects, so I can not provide a solution

**Comment by EmilAlipiev** [comment]
@ghuntley do you know about this file? I had this file in IOS and Android projects but after upgrading akavache 5 to 6-pre and sqlraw libraries yesterday, they were automatically removed from my projects. Do akavache 6  version or newest sqlite.raw libraries not need it any longer?

**Comment by jzeferino** [comment]
cc @paulcbetts

**Comment by shalinasanga** [comment]
In latest Akavache 5.0, you need to add "LinkerPreserve" class to platform projects, if its not added to Android, can not do Release build. (I add that class to all three platform specific projects)
`public static class LinkerPreserve
{
  static LinkerPreserve()
  {
    var persistentName = typeof(SQLitePersistentBlobCache).FullName;
    var encryptedName = typeof(SQLiteEncryptedBlobCache).FullName;
  }
}`
